transformer_lm/jax_main.py

```diff
@@ -9,7 +9,7 @@
 import jax
 import jax.numpy as jnp
 import numpy as np
-from tree_utils import tree_flatten
+from mlx.utils import tree_flatten
 
```

Comment:
Nit: There is a comment below that lists TODOs, maybe we can remove that (or at least the posenc bit)